services/venice-server/src/main/java/com/linkedin/venice/listener/request/MetadataFetchRequest.java

```diff
@@ -0,0 +1,44 @@
+package com.linkedin.venice.listener.request;
+
+import com.linkedin.venice.exceptions.VeniceException;
+import io.netty.handler.codec.http.HttpRequest;
+import java.net.URI;
+
+
+/**
+ * {@code MetadataFetchRequest} encapsulates a GET request to /metadata/storename on the storage node to fetch metadata
+ * for that node.
+ */
+public class MetadataFetchRequest {
+  private final String storeName;
+
+  private MetadataFetchRequest(String storeName) {
+    this.storeName = storeName;
+  }
+
+  public static MetadataFetchRequest parseGetHttpRequest(HttpRequest request) {
+    String uri = request.uri();
```

Comment:
Should probably move the first half of this function as a separate static class and call it from here and other places to avoid duplication?

Reply:
added a new class `RequestHelper` with helper `getRequestParts`